src/http-client.js

```diff
@@ -104,14 +120,26 @@ export class HttpClient {
   fetch(input: Request|string, init?: RequestInit): Promise<Response> {
     this::trackRequestStart();
 
-    let request = Promise.resolve().then(() => this::buildRequest(input, init, this.defaults));
+    let request = Promise.resolve().then(() => this::buildRequest(input, init, this.defaults))
+      .then( result => {
+        const { retryConfig } = result;
+        if (retryConfig) {
+          result.retryConfig = retryConfig;
+        }
+        return result;
+      });
+
     let promise = processRequest(request, this.interceptors)
       .then(result => {
         let response = null;
 
         if (Response.prototype.isPrototypeOf(result)) {
           response = result;
         } else if (Request.prototype.isPrototypeOf(result)) {
+          if (result.retryConfig) {
+            result.retryConfig.requestClone = result.clone();
+            result.retryConfig.client = this;
+          }
```

Comment:
Same as above. Should be in the interceptor, no need for it in the HttpClient.

Reply:
It is impossible for this code to be in the in the interceptor. The interceptor does not get a reference to the `HttpClient`.